xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/client/specifiers/LedgerSpecifier.java

```diff
@@ -0,0 +1,161 @@
+package org.xrpl.xrpl4j.model.client.specifiers;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.google.common.base.Preconditions;
+import org.immutables.value.Value;
+import org.xrpl.xrpl4j.model.jackson.modules.LedgerSpecifierDeserializer;
+import org.xrpl.xrpl4j.model.transactions.Hash256;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+/**
+ * Represents one of the three ways of specifying a ledger in a rippled API request.
+ *
+ * @see "https://xrpl.org/basic-data-types.html#specifying-ledgers"
+ */
+@Value.Immutable
+@JsonSerialize(as = ImmutableLedgerSpecifier.class)
+@JsonDeserialize(as = ImmutableLedgerSpecifier.class, using = LedgerSpecifierDeserializer.class)
+public interface LedgerSpecifier {
+
+  /**
+   * Construct a {@link LedgerSpecifier} with a ledger hash.
+   *
+   * @param ledgerHash A {@link Hash256} containing the ledger hash of the desired ledger.
+   *
+   * @return A {@link LedgerSpecifier} containing {@code ledgerHash}.
+   */
+  static LedgerSpecifier ledgerHash(Hash256 ledgerHash) {
+    return ImmutableLedgerSpecifier.builder()
+      .ledgerHash(ledgerHash)
+      .build();
+  }
+
+  /**
+   * Construct a {@link LedgerSpecifier} with a numerical ledger index.
+   *
+   * @param ledgerIndex The {@link LedgerIndex} of the desired ledger.
+   *
+   * @return A {@link LedgerSpecifier} containing {@code ledgerIndex}.
+   */
+  static LedgerSpecifier ledgerIndex(LedgerIndex ledgerIndex) {
+    return ImmutableLedgerSpecifier.builder()
+      .ledgerIndex(ledgerIndex)
+      .build();
+  }
+
+  /**
+   * Construct a {@link LedgerSpecifier} with a ledger index shortcut.
+   *
+   * @param ledgerIndexShortcut A {@link LedgerIndexShortcut} specifying a ledger index shortcut.
+   *
+   * @return A {@link LedgerSpecifier} containing {@code ledgerIndexShortcut}.
+   */
+  static LedgerSpecifier ledgerIndexShortcut(LedgerIndexShortcut ledgerIndexShortcut) {
+    return ImmutableLedgerSpecifier.builder()
+      .ledgerIndexShortcut(ledgerIndexShortcut)
+      .build();
+  }
+
+  /**
+   * A 20-byte hex string for the ledger version to use.
+   *
+   * @return An optionally-present {@link Hash256}.
+   */
+  Optional<Hash256> ledgerHash();
+
+  /**
+   * The ledger index of the ledger to use.
+   *
+   * @return An optionally-present {@link LedgerIndex}.
+   */
+  Optional<LedgerIndex> ledgerIndex();
```

Comment:
If you're deprecating `LedgerIndex`, you should probably deprecate this as well.

Reply:
Do I have to if I'm getting rid of this in #126?

Reply:
Oh, this is also the new `LedgerIndex`, not the one that's deprecated.